### PR TITLE
:tada: etl-staging-sync tool

### PR DESCRIPTION
**Pull request by Marigold**
CLI tool `etl-staging-sync` for syncing charts from source environment to target. This is especially useful for syncing work from staging servers to production.

The gist of the logic is:
- New charts are synced as **drafts** to production
- Chart updates or revisions are created as **revisions** in production (revisions could be applied directly with `--apply-revisions` flag)
- You get a warning if the chart has been updated in production

The basic usage looks like this (assuming you have properly set `.env.prod.write`)
```
etl-staging-sync staging-site-my-branch .env.prod.write --dry-run
```

See [detailed usage and logic](https://github.com/owid/etl/pull/1879/files#diff-3371c285b7e9821bab47fd2df467e89faef8c5fa4cbde9a2e324ddbbc268e28dR59)

Some questionable decisions I made:

1. Charts are created with Admin API. For that, I need to create a temporary session id in `sessions` table. It looks hacky, but works surprisingly well
2. Staging server needs to exist to sync charts. The [original idea](https://www.notion.so/2023-10-19-synchronizing-data-from-ETL-staging-servers-to-production-53d6b389f70e4aa2b3741b06bd8fa7df) was to create a new table in production database, move all charts there and kill staging server. Instead, I've decided to keep staging servers **alive for 7 days after merging** to master to have time for sync
3. It doesn't have any UI... this makes things simple for now, but it's possible that we turn it into streamlit app in the future

## TODO
* [x] remove new cherry blossom version that was used for testing
* [x] make it work for https://owid.cloud/admin

### CODE REVIEW

**Comment by pabloarosado** [comment]
Hey @Marigold I just tested this tool with the temporary server of [this PR](https://github.com/owid/etl/pull/1893) and it worked well, thanks for this!

File: dag/biodiversity.yml [hunk 1]
```diff
@@ -5,4 +5,8 @@
     data://garden/biodiversity/2023-01-11/cherry_blossom:
     - data://meadow/biodiversity/2023-01-11/cherry_blossom
     data://grapher/biodiversity/2023-01-11/cherry_blossom:
-    - data://garden/biodiversity/2023-01-11/cherry_blossom
\ No newline at end of file
+    - data://garden/biodiversity/2023-01-11/cherry_blossom
+
+
+    data://grapher/biodiversity/2023-10-10/cherry_blossom:
```
Comment (pabloarosado) [inline dag/biodiversity.yml]: To be removed.

File: etl/steps/data/grapher/biodiversity/2023-10-10/cherry_blossom.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+"""Load a garden dataset and create a grapher dataset."""
```
Comment (pabloarosado) [inline etl/steps/data/grapher/biodiversity/2023-10-10/cherry_blossom.py]: To be removed.

File: apps/staging_sync/cli.py [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+    help="Sync only this chart id.",
+)
+@click.option(
+    "--publish/--no-publish",
+    default=False,
+    type=bool,
+    help="Automatically publish new charts.",
+)
+@click.option(
+    "--apply-revisions/--no-apply-revisions",
```
Comment (pabloarosado) [inline apps/staging_sync/cli.py:38]: Shouldn't it be called "--approve-revisions/--keep-revisions"?

File: apps/staging_sync/cli.py [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+    source: Path,
+    target: Path,
+    chart_id: Optional[int],
+    publish: bool,
+    apply_revisions: bool,
+    dry_run: bool,
+) -> None:
+    """Syncs grapher charts and revisions modified by Admin user from source_env to target_env. (Admin user is used by staging servers).
+
+    SOURCE and TARGET can be either paths to .env file or name of a staging server.
```
Comment (pabloarosado) [inline apps/staging_sync/cli.py:61]: Would it be possible to use here `master` as a target?

**Review by pabloarosado — APPROVED** [review]
It looks good, and I didn't find any issues when running it.

File: apps/staging_sync/cli.py (hunk 4 above)
Comment (Marigold) [reply apps/staging_sync/cli.py:61]: Using `master` would sync them to `staging-sync-master` (which exists and is being cloned to new staging servers).

Writing to production shouldn't be too easy. It's ok for super users, but we might build some simple UI for others with unique MySQL user that can only write to selected tables.

**Pull request merged**